src/microbe_stage/systems/EngulfingSystem.cs

```diff
@@ -43,12 +43,14 @@
 [WritesToComponent(typeof(MicrobeColony))]
 [WritesToComponent(typeof(MicrobeAI))]
 [WritesToComponent(typeof(TemporaryEndosymbiontInfo))]
+[WritesToComponent(typeof(DamageCooldown))]
 [ReadsComponent(typeof(CollisionManagement))]
 [ReadsComponent(typeof(MicrobePhysicsExtraData))]
 [ReadsComponent(typeof(OrganelleContainer))]
 [ReadsComponent(typeof(MicrobeEventCallbacks))]
 [ReadsComponent(typeof(WorldPosition))]
 [ReadsComponent(typeof(EntityRadiusInfo))]
+[ReadsComponent(typeof(DamageCooldown))]
```

Comment:
This is a good start but I have a few comments.

This attribute shouldn't be here as this writes to the damage cooldown component (write implies also reading).

Reply:
Do you mean `[WritesToComponent(typeof(DamageCooldown))]`? I'm just learning the `GenerateThreadedSystems` class. It can take some time.

I added these two attributes because I get the damage cooldown information for the entity and then I modify it. So I thought I need both: reading and writing.

Reply:
Yes, if there is a WritesTo attribute that means there should not be ReadsFrom attribute for the same component type. Writing already implies reading in the system.